server/app.js

```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
+                    client.write(buf);
+
+                    client.destroy();
+                    client = new net.Socket();
+                    client.on('err', function(err){
+                        // handle the error safely
+                        console.log(err)
+                    });
+                });        
+        }
+        return 0;
+
+    })
+    //output
+}
+
+function looking_for_path() {
+    current_state = state[1];
+
+    console.log("HLooking for path");
+                //STOP
+                
+                client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+                    console.log('Connected to 192.168.43.250'); 
+                    console.log('Local client sending:', 0);
+                    buf = new Buffer([14, 0x4, 0xFF, 18, 1, 0xFF]); // 0x4B is temp value 75
+                    client.write(buf);
+                    client.destroy();
+                    client = new net.Socket();
+                    client.on('err', function(err){
+                        // handle the error safely
+                        console.log(err)
+                    });
+                });  
+                setTimeout(found_path_to_house, 1000);                 
+
+}
+
+function found_path_to_house() {
+    current_state = state[2];
+
+    console.log("found path to house");
+                    client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+                    console.log('Connected to 192.168.43.250'); 
+                    console.log('Local client sending:', 0);
+                    buf = new Buffer([17, 0x4, 0xFF, 18, 2, 0xFF]); // 0x4B is temp value 75
+                    client.write(buf);
+                    client.destroy();
+                    client = new net.Socket();
+                    client.on('err', function(err){
+                        // handle the error safely
+                        console.log(err)
+                    });
+                });  
+                setTimeout(navigating_path, 1000);      
+    // Pivot 90 degrees and go down new path
+}
+
+
+function navigating_path() {
+    current_state = state[4];
+    console.log("navigate path");
+                    client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+                    console.log('Connected to 192.168.43.250'); 
+                    console.log('Local client sending:', 0);
+                    buf = new Buffer([0,0x32, 0xFF, 18, 3, 0xFF]);
+                    client.write(buf);
+                    client.destroy();
+                    client = new net.Socket();
+                    client.on('err', function(err){
+                        // handle the error safely
+                        console.log(err)
+                    });
+                });  
+
+
+    console.log("navigating_path");
+    // continue until sensors reach door
+}
+
+function reached_door() {
+    current_state = state[5];
```

Comment:
Instead of `state[5]` which is non-intuitive, there should be a way of explicitly selecting the state.
In C what I would do is `state[DELIVER_PACKAGE]` to specify that i want the index in the array of state that corresponds to 5=DELIVER_PACKAGE.
In Python what I would do is `state["DELIVER_PACKAGE"]` to search in the dictionary for the state.